### PR TITLE
LibWasm: more instructions

### PR DESCRIPTION
**Pull request by BrandonKi**
Implements a few random missing instructions.

### CODE REVIEW

**Comment by alimpfard** [comment]
Oh nice, I'll remove my `popcnt` impl from #7670.

File: Userland/Libraries/LibWasm/AbstractMachine/Interpreter.cpp [hunk 1]
```diff
@@ -707,11 +730,15 @@ void BytecodeInterpreter::interpret(Configuration& configuration, InstructionPoi
...
+        BINARY_PREFIX_NUMERIC_OPERATION(u32, rotl, i32);
     case Instructions::i32_rotr.value():
+        BINARY_PREFIX_NUMERIC_OPERATION(u32, rotr, i32);
     case Instructions::i64_clz.value():
+        UNARY_NUMERIC_OPERATION(i64, __builtin_clzl);
     case Instructions::i64_ctz.value():
+        UNARY_NUMERIC_OPERATION(i64, __builtin_ctzl);
     case Instructions::i64_popcnt.value():
-        goto unimplemented;
+        UNARY_NUMERIC_OPERATION(i64, __builtin_popcountl);
```
Comment (tomuta) [inline Userland/Libraries/LibWasm/AbstractMachine/Interpreter.cpp:741]: Shouldn't this use `__builtin_popcountll`?

**Review by tomuta — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibWasm/AbstractMachine/Interpreter.cpp (hunk 1 above)
Comment (BrandonKi) [reply Userland/Libraries/LibWasm/AbstractMachine/Interpreter.cpp:741]: Yea you're right. I'll fix that.

**Comment by alimpfard** [comment]
Putting it in a separate function and printing the inputs out made it spit out 31 instead (probably related to the return type of `__builtin_clz`)
```
clz -1 = 0
clz 0 = 31
clz 32768 = 16
clz 255 = 24
clz -2147483648 = 0
clz 1 = 31
clz 2 = 30
clz 2147483647 = 1
```

```
         Test:   execution of i32_0: clz (instance 1) (failed):
                 ExpectationError: toBe: expected _32_, got _31_
```

**Comment by BrandonKi** [comment]
It's fixed now. Apparently the result of __builtin_clz is undefined when the operand is 0.

File: Userland/Libraries/LibWasm/AbstractMachine/Interpreter.cpp [hunk 2]
```diff
@@ -368,6 +368,34 @@ ALWAYS_INLINE static T rotr(T value, R shift)
...
+
+template<typename T>
+ALWAYS_INLINE static i32 ctz(T value)
+{
+    if (value == 0)
+        return sizeof(T) * CHAR_BIT;
+
+    if constexpr (sizeof(T) == 32)
+        return __builtin_ctz(value);
+    else if constexpr (sizeof(T) == 64)
```
Comment (alimpfard) [inline Userland/Libraries/LibWasm/AbstractMachine/Interpreter.cpp:393]: `sizeof(i64)` is 8, not 64.

File: Userland/Libraries/LibWasm/AbstractMachine/Interpreter.cpp [hunk 3]
```diff
@@ -368,6 +368,34 @@ ALWAYS_INLINE static T rotr(T value, R shift)
...
+        VERIFY_NOT_REACHED();
+}
+
+template<typename T>
+ALWAYS_INLINE static i32 ctz(T value)
+{
+    if (value == 0)
+        return sizeof(T) * CHAR_BIT;
+
+    if constexpr (sizeof(T) == 32)
```
Comment (alimpfard) [inline Userland/Libraries/LibWasm/AbstractMachine/Interpreter.cpp:391]: Same, `sizeof(i32)` is 4, not 32.

(the build will probably fail because of this)

**Pull request merged**